pkg/cloud/cloud.go

```diff
@@ -247,3 +320,24 @@ func parseEfsTags(tagMap map[string]string) []*efs.Tag {
 	}
 	return efsTags
 }
+
+func getAvailableMountTargets(mountTargets []*efs.MountTargetDescription) []*efs.MountTargetDescription {
+	availableMountTargets := []*efs.MountTargetDescription{}
+	for _, mt := range mountTargets {
+		if *mt.LifeCycleState == "available" {
+			availableMountTargets = append(availableMountTargets, mt)
+		}
+	}
+
+	return availableMountTargets
+}
+
+func getMountTargetForAz(mountTargets []*efs.MountTargetDescription, azName string) *efs.MountTargetDescription {
+	for _, mt := range mountTargets {
+		if *mt.AvailabilityZoneName == azName {
+			return mt
+		}
+	}
+	klog.Infof("There is no mount target match %v. Will pick a random mount target.", azName)
+	return nil
+}
```

Comment:
Any idea how we can avoid to implement the same logic on both efs-utils side and csi-driver side?

Reply:
One possible way is for efs-utils to accept `role` as a mount option.

Reply:
I guess comparing with this approach, adding a role option should be preferred, that will reduce the dup logic in the driver side, also on efs-utils side we provide the output & log message like the ip cannot be connected yet, no available mount target in the AZ, we can reuse it here.

So is this a temp workaround? Should we deprecate the duplicates once the role arn option can be used?

Reply:
While `role` mount option is preferred to reduce the dup logic, I'm not sure if csi-driver can capture all the messages from efs-utils. CSI driver only captures the message(s) shown on the console after mount attempt and not messages logged in `mount.log`. I feel it is an overhead to reach mount.log and would rather throw the messages in the driver logs.

What do you think?

Reply:
The mount attempt will output the message to the console, e.g. https://github.com/aws/efs-utils/blob/master/src/mount_efs/__init__.py#L1789-L1790 here we use fatal_error to throw back and output the error message.

It will be hard if we change the common logic in two place in the future, also if we add a logic, we need to implement the same logic in the driver, that will be not easy to maintain, and csi-driver should be mainly responsible for managing. 

So IMO for this feature we can use the logic here, once we add the role feature, we should explore to optimize the code path and reduce the common logic as much as possible. Let efs-utils handle most of the mount work and logic.